### PR TITLE
upgrade gulp and composer updates

### PR DESCRIPTION
**Pull request by chrisherold**
This upgrades gulp to v4 and updates all packages. The gulp file is renamed to `gulpfile.babel.js` and uses `@babel/register` and a `.babelrc` file to tell gulp to compile the ES6. It adds a step in the scripts task to compile any ES6 JS in the `assets/js` using Babel. The `.nvmrc` is set to the current version of Node 12. 

Also bumped all composer packages as well.

### CODE REVIEW

**Comment by yuvilio2** [comment]
Thanks @chrisherold . Gave this a go. A few early impressions.

* Installing went smooth
* Composer wordpress plugin updates seems fine (thanks for bumping the PHP version to what modern hosts are using).
* I tried to live edit a stylesheet in two ways. 
  - First I tried an invalid style "back", but didn't see an error on the terminal log, when i saved. 
  - Second, I tried a valid style, `background-color: black` on the body class of a page. but the browser didn't reflect the change, automatically.  It only did when i manually refreshed the browser page. 


Peek 2020-04-28 11-02

Can someone else confirm these issues? Or are they mine specifically?

My local setup: 

```bash
$ gulp --version
CLI version: 2.2.0
Local version: 4.0.2
$ node --version
v12.16.2
$ ./node_modules/gulp/bin/gulp.js --version
CLI version: 2.2.0
Local version: 4.0.2
$ ./node_modules/browser-sync
browser-sync/        browser-sync-client/ browser-sync-ui/
$ ./node_modules/browser-sync/dist/bin.js --version
2.26.7
$ 

```

**Comment by yuvilio** [comment]
Got a chance to troubleshoot the issue i raised above and have a PR, #120 , into this PR's branch,  `new-gulp`. Seems to fix the scss watch browsersync not injecting the updated css, and still using the stream api rather than doing a full browser refresh.  More details i that PR

@chrisherold , feel free to look that PR over and merge if looks good.  I think after that fix, this gulp upgrade might be good to go, unless someone else wants to have a look.

**Comment by yuvilio** [comment]
Just tested this branch `new-gulp` with @chrisherold 's updates in e5ba5a51 . Gulp watch scss updates (stream injected) is now working fine. Gulp watch js is as well for normal reloads. the dist/ directory seems to generate fine, and @chrisherold  tested a deploy with it to a staging site on one of our project. 

So this upgraded gulp 4 setup looks fine to me to merge to master and start using, unless anyone else wants to test.

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-8
+12.16.2
```
Comment (ccorda) [inline .nvmrc:1]: Can we set this to 12? How we've always done it, ensuring that whatever version of 12 the dev is using is sufficient.

File: composer.json [hunk 2]
```diff
@@ -4,98 +4,97 @@
         "vendor-dir": "wp-content/plugins/composer-libs"
     },
     "require": {
-        "php": ">=5.4.0",
-        "WordPress/WordPress": "5.0.3",
-        "advanced-custom-fields/advanced-custom-fields-pro": "5.7.11",
-        "norcross/airplane-mode": "0.2.2",
-        "roots/soil": "3.7.3",
-        "upstatement/routes": "0.4.0",
+        "php": ">=7.3.16",
```
Comment (ccorda) [inline composer.json:7]: can we set this as 7.3.0? i have version 7.3.11, that should be sufficient without having to go off on php upgrades.

Comment (chrisherold) [reply composer.json]: done -- 43a00f98

File: .nvmrc (hunk 1 above)
Comment (chrisherold) [reply .nvmrc]: done -- 43a00f98

**Pull request merged**